### PR TITLE
refactor(createProject): add types and remove useless projectForm arg…

### PR DESCRIPTION
**Pull request by joaoneto**
…s iteration

# Pull Request Guidelines

Our guidelines for submitting a pull request.

## Before submitting a Pull Request, please make sure you have verified the following:

- [ ] The commit message follows our guidelines:
  - A good commit message should be two things: meaningful and concise. It should not contain every single detail, describing each changed line—we can see all the changes in Git—but, at the same time, it should say enough to avoid ambiguity.
  - We use Microverse's commit message convention
  - The convention stablish that a commit message has to be in the present tense, imperative and lowercase.
  - Example: `fix typo in README.md`
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Please describe the current behavior that you are modifying, or link to a relevant issue.

Issue Number: N/A

## What is the new behavior?

Describe the new behavior or link to a relevant issue.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications below.

## Other information

Since the projectForm already sends in the order of the arguments correctly, to set the variables, it is not necessary to use an iteration, just use the deconstructure

### CODE REVIEW

**Comment by rsaz** [comment]
@Daniel-Boll @juliano-soares please review the PR,:

1 - Test with npm pack and install it globally
2 - Test on windows and wsl
3 - Verify the scenarios in which the change has code impact, scaffold, command responses.

Waiting your return for approval and publishing

**Review by juliano-soares — APPROVED** [review]
Great job!!

File: src/new/form.ts [hunk 1]
```diff
@@ -86,25 +86,23 @@ const enum PackageManager {
...
+];
+
+const projectForm = async (projectName: string, args: ProjectFormArgs): Promise<void> => {
+	let answer: {
+		name: string;
+		packageManager: string;
+		template: Template;
+		confirm: boolean;
+	};
 	const projName: string = projectName;
```
Comment (Daniel-Boll) [inline src/new/form.ts:104]: It is not related to your changes, but thinking enough about it, this variable is useless this way, we can already use the `projectName` it self, can you add this change as well?

**Review by Daniel-Boll — APPROVED** [review]

**Review by Daniel-Boll — APPROVED** [review]
LGTM

**Review by rsaz — CHANGES_REQUESTED** [review]
Removing the argument order you remove the option of the user silently install the application at any order. Your code change only allow users to create projects in this form:

`expressots new app-demo -p npm -t opinionated`

Our current solution allows the user to pass the -p (package manager) and -t (template) at any order

**Review by rsaz — APPROVED** [review]

**Pull request merged**